bip-0324.mediawiki

```diff
@@ -0,0 +1,467 @@
+
+
+=== NODE_P2P_V2 ===
+
+Peers supporting the transport protocol after this proposal MUST signal <code>NODE_P2P_V2</code>
+<pre>
+NODE_P2P_V2 = (1 << 11)
+</pre>
+
+A peer usually learns an address along with the expected service flags which MAY be used to filter possible outbound peers.
+
+A peer signaling <code>NODE_P2P_V2</code> MUST accept encrypted communication specified in this proposal.
+
+Peers MAY only make outbound connections to peers supporting <code>NODE_P2P_V2</code>.
+
+=== Handshake ===
+
+<pre>
+ ----------------------------------------------------------------------------------------
+ | Initiator                             Responder                                      |
+ |                                                                                      |
+ | x, X         := SECP256k1_KEYGEN()                                                   |
+ | CLIENT_HDATA := X                                                                    |
+ |                                                                                      |
+ |               --- CLIENT_HDATA --->                                                  |
+ |                                                                                      |
+ |                                       y, Y           := SECP256k1_KEYGEN()           |
+ |                                       ECDH_KEY       := SECP256k1_ECDH(X,y)          |
+ |                                       SERVER_HDATA   := Y                            |
+ |                                                                                      |
+ |               <-- SERVER_HDATA ----                                                  |
+ |                                                                                      |
+ | ECDH_KEY     := SECP256k1_ECDH(x,Y)                                                  |
+ ----------------------------------------------------------------------------------------
+</pre>
+
+To request encrypted communication (only possible if yet no other messages have been sent or received), the initiating peer generates an EC secp256k1 ephemeral key and sends the corresponding 32-byte public key to the responding peer and waits for the remote 32-byte public key from the counterparty.
+
+ODD secp256k1 public keys MUST be used (public keys starting with 0x02). If the public key from the generated ephemeral key is an EVEN public key (starting with 0x03), its public key SHOULD be negated and then recalculated.
+Only using ODD public keys makes it more complex to identify the handshake based on analyzing the traffic.
+
+The handshake request and response message are raw 32byte payloads containing no header, length or checksum (the pure 32byte payload) and MUST be sent before anything else.
+
+Public keys starting with the 4-byte network magic are forbidden and MUST lead to local regeneration of an ephemeral-key.
```

Comment:
here you hyphenate ephemeral-key, but above it is written without the hyphen but with a space